br/pkg/utils/db.go

```diff
@@ -30,3 +35,49 @@ type DBExecutor interface {
 	StmtExecutor
 	BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx, error)
 }
+
+// CheckLogBackupEnabled checks if LogBackup is enabled in cluster.
+// this mainly used in three places.
+// 1. Resolve locks to scan more locks after safepoint.
+// 2. Add index skipping use lightning.
+// 3. Telemetry of log backup feature usage statistics every 6 hours.
+// NOTE: this result shouldn't be cached by caller. because it may change every time in one cluster.
+func CheckLogBackupEnabled(ctx sessionctx.Context) bool {
+	enabled, err := IsLogBackupEnabled(ctx.(sqlexec.RestrictedSQLExecutor))
```

Comment:
if we always assert the session context to this type, why not accept `sqlexec.RestrictedSQLExecutor` directly?

(Or we need to check whether the check successes, returning `true` when failed to make the assertion?)

Reply:
These three invoke can tolerate false positive when we get error(maybe internet error)

Reply:
Type assertion failure(without checking) can lead to panic, I'm not sure whether this is acceptable.